index.js

```diff
@@ -108,31 +133,28 @@ export function activate() {
 
 	this.subscriptions = new CompositeDisposable();
 	this.subscriptions.add(atom.commands.add('atom-text-editor', {
-		'XO:Fix': () => {
-			const editor = atom.workspace.getActiveTextEditor();
-
-			if (!editor) {
-				return;
-			}
-
-			let report;
-
-			allowUnsafeNewFunction(() => {
-				report = lintText(editor.getText(), {
-					fix: true,
-					cwd: path.dirname(editor.getPath())
-				});
-			});
+		'XO:Fix': fix
+	}));
 
-			const output = report.results[0].output;
+	atom.workspace.observeTextEditors(editor => {
+		editor.getBuffer().onWillSave(() => {
+			const isJS = SUPPORTED_SCOPES.includes(editor.getGrammar().scopeName);
+			const shouldFixOnSave = atom.config.get('linter-xo.fixOnSave');
 
-			if (output) {
-				setText(output);
+			if (isJS && shouldFixOnSave) {
+				fix();
```

Comment:
Pass `editor` to `fix()` instead of getting the active editor inside the `fix` function. You also have to change line `136` to a function that get's the active editor.